src/committee/orchestrator.rs

```diff
@@ -75,79 +83,120 @@ impl Orchestrator {
             )
             .await
-            .context("rpc request to committee didn't work");
-            debug!("{:?}", resp);
-            let resp = resp?;
-
-            let response: bitcoincore_rpc::jsonrpc::Response = serde_json::from_str(&resp)?;
-            let resp: Round1Response = response.result()?;
+            {
+                Ok(x) => x,
+                Err(err) => {
+                    info!("Error: member {member_id:?} in round 1 sending json rpc request: {err}");
+                    continue;
+                }
+            };
+
+            // parse the response json rpc type
+            let round1_parse_json_rpc_response: bitcoincore_rpc::jsonrpc::Response =
+                match serde_json::from_str(&round1_send_json_rpc_resp) {
+                    Ok(x) => x,
+                    Err(err) => {
+                        info!(
+                        "Error: member {member_id:?} in round 1 parsing json rpc response: {err}"
+                    );
+                        continue;
+                    }
+                };
+
+            // parse the actual response
+            let round1_parse_actual_response: Round1Response = match round1_parse_json_rpc_response
+                .result()
+            {
+                Ok(x) => x,
+                Err(err) => {
+                    info!("Error: member {member_id:?} in round 1 parsing actual response: {err}");
+                    continue;
+                }
+            };
 
             // store the commitment
-            commitments_map.insert(**member_id, resp.commitments);
+            results.lock().await.push((
+                member_id,
+                member,
+                round1_parse_actual_response.commitments,
+            ));
         }
 
         //
         // Produce transaction and digest
         //
         let message = get_digest_to_hash(&bob_request.prev_outs, &bob_request.tx, &smart_contract)?;
+        let commitments_map: BTreeMap<_, _> = results
+            .lock()
+            .await
+            .iter()
+            .map(|(member_id, _, commitments)| (***member_id, commitments.clone()))
+            .collect();
 
         //
         // Round 2
         //
 
         let mut signature_shares = BTreeMap::new();
 
+        // Preparing for round 2
         let round2_request = Round2Request {
             txid: bob_request.txid()?,
             proof_hash: bob_request.proof.hash(),
             commitments_map: commitments_map.clone(),
             message,
         };
 
-        // TODO: do this concurrently with async
-        // TODO: take a random sample instead of the first `threshold` members
-        // TODO: what if we get a timeout or can't meet that threshold? loop? send to more members?
-        for (member_id, member) in &threshold_of_members {
+        // Concurrently send round 2 requests
+        for (member_id, member, _) in results.lock().await.iter() {
```

Comment:
going through this iterator should be concurrent as well, like we did for round 1